Comments page

motuncoded opened the pull request:
In this change, I create a navbar component with navigating pages and I also create the a simple scroll of comments in the commenting page

closes #11

Commit:
Create a simple scroll of posts

In this feature, I used Posts Placeholder Api which  render a simple scroll of different posts.

Commit:
Setup the prettier and made file changes

In this change, I set up prettier in the project and move the function types outside the Reactjs function

Commit:
Create an infinite scroll of posts

In this change, I created a Posts.tsx file that can allow infinite scroll of posts with loading when navigatimg down the posts.

Commit:
Create an infinite scroll of posts

In this change, I created a Posts.tsx file that can allow infinite scroll of posts with loading when navigatimg down the posts.

motuncoded committed:
Merge branch 'main' into infinite-scroll-of-posts

Commit:
Fix the infinite scroll of posts

In this change I fixed the changes based on the review to have an infinite scroll of posts

Commit:
Merge branch 'infinite-scroll-of-posts' of https://github.com/motuncoded/social_website into infinite-scroll-of-posts

Commit:
optimising next build

In this change I optimised the project by runnning the nex build command to be able to check the vercel preview

Commit:
adjust the files

In this change, I removed the posts.tsx and incorporated the rendering of the API inside the index.tsx file to avoid rendering the same thing

Commit:
Fix the problem of the loading state

In this change, I made the loading state to render the right jsx elements for loading and no more posts

Commit:
Create a functional like button

In this change I created a like button that when clicked changes the number of likes it has

motuncoded committed:
Merge branch 'main' into like-button

Commit:
fix the like button and loading state of the posts

In this change I removed the liked type and also change th loading state of the posts

Commit:
fix the like button and loading state of the posts

In this change I removed the liked type and also change th loading state of the posts

Commit:
fix the like button and loading state of the posts

In this change I removed the liked type and also change th loading state of the posts

Commit:
fix the like button and loading state of the posts

In this change I removed the liked type and also change th loading state of the posts

Commit:
fix the like button and loading state of the posts

In this change I removed the liked type and also change th loading state of the posts

Commit:
added the else statement of the fetchPosts

In this change I added the else statement in the ftchPosts

Commit:
Create the comment page

In this change, I create a navbar component with navigating pages and I also create the a simle scroll of comments in the commenting page

Commit:
Create the comment page

In this change, I create a navbar component with navigating pages and I also create the a simle scroll of comments in the commenting page

motuncoded committed:
Merge branch 'main' into comments_page

Commit:
Create the comment page

In this change, I create a navbar component with navigating pages and I also create the a simle scroll of comments in the commenting page

Commit:
Merge branch 'comments_page' of https://github.com/motuncoded/social_website into comments_page

Commit:
Create the comment page

In this change, I create a navbar component with navigating pages and I also create the a simle scroll of comments in the commenting page

Asylcreek commented on package.json:
```diff
@@ -9,6 +9,9 @@
   "dependencies": {
+    "@tanstack/react-query": "^5.45.1",
+    "@tanstack/react-query-devtools": "^5.45.1",
+    "@tanstack/react-query-next-experimental": "^5.45.1",
```
Please remove this, you do not seem to be using it

Asylcreek commented on pages/commenting.tsx:
```diff
@@ -0,0 +1,61 @@
+};
+
+function Commenting() {
+  type Comment = {
+    id: number;
+    body: string;
+    name: string;
+    email: string;
+  };
```
This should be outside of the React function

Asylcreek commented on pages/commenting.tsx:
```diff
@@ -0,0 +1,61 @@
+    email: string;
+  };
+  const { data, error, isLoading } = useQuery({
+    queryKey: ["comments"],
```
You should have all your keys in a central place

motuncoded replied:
I dont understand this

Asylcreek commented on pages/commenting.tsx:
```diff
@@ -0,0 +1,61 @@
+  return res.json();
+};
+
+function Commenting() {
```
```suggestion
function CommentPage() {
```

Asylcreek commented on pages/commenting.tsx:
Please rename the file to `comments.tsx`

Asylcreek commented on pages/index.tsx:
```diff
@@ -1,151 +1,15 @@
-type Error = {
-  message: string;
-};
+import React from "react";
```
You do not need to import React

Asylcreek commented on pages/posting.tsx:
Please rename the file to `posts.tsx`

Asylcreek commented on pages/posting.tsx:
```diff
@@ -0,0 +1,149 @@
+  const [hasMore, setHasMore] = useState(false);
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+
+  const fetchPosts = async () => {
```
You should convert this `fetch` to use `react-query`

Asylcreek commented on pages/users.tsx:
```diff
@@ -0,0 +1,7 @@
+import React from "react";
```
You do not need to import React

Asylcreek replied:
React is still here ma

Commit:
fix the type conversion in Comments page

In this change, I fixed some page naming convention and move their type deinition outside the function

Asylcreek commented on pages/comments.tsx:
```diff
@@ -0,0 +1,61 @@
+};
+function Comments() {
+  const { data, error, isLoading } = useQuery({
+    queryKey: ["comments"],
```
Instead of `["comments"]`, export all your query keys from another file and use the imported value here

Commit:
export the query keys

In this change, I create the query file that export the query keys outside the React function

Asylcreek commented on pages/query.ts:
```diff
@@ -0,0 +1,16 @@
+};
+export const useComments = () => {
+  const { data, error, isLoading } = useQuery({
+    queryKey: ["comments"],
```
I still want you to create a file where you export all your queryKeys from.

So maybe a `queryKeys.ts` file that would export an object that has `QUERY_KEY: 'query_key'` pairs, for example
```js
const queryKeys = {
COMMENT: 'comment'
}
```

This would serve as a source of truth for every query key you would be using, even you mutation keys.

motuncoded replied:
ok

Commit:
create the queryKeys file

In this change, I created a queryKeys.ts file to export the querykeys as objects and also deleted some unnecssary packages

Asylcreek commented on pages/comments.tsx:
```diff
@@ -0,0 +1,60 @@
+  return res.json();
+};
+function Comments() {
+  const { data, error, isLoading } = useQuery({
+    queryKey: [queryKeys.COMMENT],
+    queryFn: getComments,
+  });
```
You were on the right track creating a separate hook for this. I did not ask you to remove that, I only asked you to incorporate the `queryKeys` thing into it

motuncoded replied:
ok

Asylcreek commented:
And your deployment failed, please find out why

Commit:
updated the queryKeys file and the hoooks

In this change, I updated the queryKeys file and the comments hook

Commit:
updated the queryKeys file and the hoooks

In this change, I updated the queryKeys file and the comments hook

Asylcreek commented on pages/comments.tsx:
```diff
@@ -0,0 +1,52 @@
+import React from "react";
```
You do not need to import React

Asylcreek commented on pages/components/Navbar.tsx:
```diff
@@ -0,0 +1,21 @@
+import Link from "next/link";
+import React from "react";
```
You do not have to import React

Commit:
Removed all the import react

In this change, I removed the react import from the files
